### PR TITLE
Deduplication of fingerprints (and other fixes)

### LINKED ISSUE #338: Deduplication between systems from multiple data source inspection
## Specify type:
- Enhancement

## Description:
Systems between Network and VCenter data sources should be deduplicated via:
- dmidecode bios uuid vs. vm uuid
- mac address

### PR DESCRIPTION
**Pull request by kholdaway**
- Remove duplicate fingerprints from network scan
- Merge network and vcenter scan
- Normalized JSON in fingerprints for better display in django UI
- Added Vcenter facts to fingerprint results

Closes #338

### CODE REVIEW

File: quipucords/api/fingerprint/model.py [hunk 1]
```diff
@@ -31,20 +31,36 @@ class SystemFingerprint(models.Model):
...
+    ip_addresses = models.CharField(max_length=1024, unique=False, null=True)
 
     cpu_count = models.PositiveIntegerField(unique=False, null=True)
+
+    # Network scan facts
+    bios_uuid = models.CharField(max_length=36, unique=False, null=True)
+    subscription_manager_id = models.CharField(
+        max_length=36, unique=False, null=True)
+
+    connection_uuid = models.CharField(max_length=36, unique=False, null=True)
```
Comment (chambridge) [inline quipucords/api/fingerprint/model.py]: I don't think we need the `connection_*` facts in the fingerprint.

File: quipucords/api/fingerprint/model.py [hunk 2]
```diff
@@ -53,71 +69,98 @@ class SystemFingerprint(models.Model):
...
-    virtualized_host_socket_count = models.PositiveIntegerField(unique=False,
-                                                                null=True)
-    virtualized_cluster = models.CharField(max_length=128,
-                                           unique=False,
-                                           null=True)
-    virtualized_datacenter = models.CharField(max_length=128,
-                                              unique=False,
-                                              null=True)
+    # VCenter scan facts
+    vm_name = models.CharField(max_length=256, unique=False, null=True)
```
Comment (chambridge) [inline quipucords/api/fingerprint/model.py]: I think maybe we should merge `connection_host` and `vm_name` to `name` field.

File: quipucords/fingerprinter/__init__.py [hunk 3]
```diff
@@ -32,203 +52,483 @@ def process_fact_collection(sender, instance, **kwargs):
...
-            elif fact.get('virt_type'):
-                fingerprint['infrastructure_type'] = 'virtualized'
+    network_fingerprints = []
+    vcenter_fingerprints = []
+    for source in raw_facts['sources']:
+        source_fingerprints = _process_source(
+            raw_facts['fact_collection_id'],
+            source)
+        if source['source_type'] == Source.NETWORK_SOURCE_TYPE:
+            network_fingerprints = network_fingerprints +\
```
Comment (chambridge) [inline quipucords/fingerprinter/__init__.py]: can't you just do `network_fingerprints += source_fingerprints`?

File: quipucords/fingerprinter/__init__.py [hunk 4]
```diff
@@ -12,15 +12,35 @@
...
 import django.dispatch
 from api.fact.util import read_raw_facts
-from api.models import FactCollection
+from api.models import FactCollection, Source
 from api.serializers import FingerprintSerializer
 
 logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
 
-# pylint: disable=unused-argument
+NETWORK_IDENTIFICATION_KEYS = ['subscription_manager_id',
```
Comment (chambridge) [inline quipucords/fingerprinter/__init__.py]: Do we need a `VCENTER_IDENTIFICATION_KEYS` for dedup with `vm_uuid`? Are we doing this with another issue? If so, can you open the issue?

File: quipucords/fingerprinter/__init__.py (hunk 3 above)
Comment (kholdaway) [reply quipucords/fingerprinter/__init__.py]: Good point

File: quipucords/fingerprinter/__init__.py (hunk 4 above)
Comment (kholdaway) [reply quipucords/fingerprinter/__init__.py]: Added and there is de-dup code

File: quipucords/api/fingerprint/model.py (hunk 2 above)
Comment (kholdaway) [reply quipucords/api/fingerprint/model.py]: Done

File: quipucords/api/fingerprint/model.py (hunk 1 above)
Comment (kholdaway) [reply quipucords/api/fingerprint/model.py]: Done

**Pull request merged**